src/test/java/lambda/part3/exercise/Mapping.java

```diff
@@ -165,7 +163,189 @@ public LazyFlatMapHelper(List<T> list, Function<T, List<R>> function) {
         }
     }
 
+    interface ReachIterable<T> {
+        boolean tryGet(Consumer<T> consumer);
+
+        static<T> ReachIterable<T> from(List<T> list) {
+            final Iterator<T> iterator = list.iterator();
+            return new ReachIterable<T>() {
+                @Override
+                public boolean tryGet(Consumer<T> consumer) {
+                    if (iterator.hasNext()) {
+                        consumer.accept(iterator.next());
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+            };
+        }
+
+        default List<T> force() {
+            final List<T> result = new ArrayList<>();
+            while (tryGet(result::add));
+            return result;
+        }
+
+        default ReachIterable<T> filter(Predicate<T> predicate) {
+            final ReachIterable<T> self = this;
+            return new ReachIterable<T>() {
+                @Override
+                public boolean tryGet(Consumer<T> consumer) {
+                    return self.tryGet(t -> {

```

Comment:
Incorrect. You should either return `false` or return `true` and call `consumer`.